File: app/src/main/java/com/softartdev/noteroom/db/DbStore.java
package com.softartdev.noteroom.db;

import android.text.Editable;

import com.softartdev.noteroom.model.Note;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface DbStore {
    List<Note> getNotes();

    Long createNote(String title, String text);

    void saveNote(long id, String title, String text);

    Note loadNote(long noteId);

    void deleteNote(long id);

    boolean checkPass(Editable pass);

    boolean isEncryption();

    void changePass(@Nullable Editable oldPass, @Nullable Editable newPass);

    boolean isChanged(long id, @NotNull String title, @NotNull String text);

    boolean isEmpty(long id);
}

File: app/src/main/java/com/softartdev/noteroom/db/DbStore.kt
package com.softartdev.noteroom.db

import android.text.Editable

import com.softartdev.noteroom.model.Note
import io.reactivex.Completable
import io.reactivex.Flowable

import io.reactivex.Maybe
import io.reactivex.Single

interface DbStore {
    val notes: Flowable<List<Note>>

    val isEncryption: Boolean

    fun createNote(title: String, text: String): Single<Long>

    fun saveNote(id: Long, title: String, text: String): Single<Int>

    fun updateTitle(id: Long, title: String): Completable

    fun loadNote(noteId: Long): Maybe<Note>

    fun deleteNote(id: Long): Single<Int>

    fun checkPass(pass: Editable): Single<Boolean>

    fun changePass(oldPass: Editable?, newPass: Editable?)

    fun isChanged(id: Long, title: String, text: String): Single<Boolean>

    fun isEmpty(id: Long): Single<Boolean>
}
